designs/2022-supress-ignored-file-warnings/README.md

```diff
@@ -0,0 +1,174 @@
+
+For example, with
+
+```
+// .eslintignore
+foo.js
+```
+
+Running
+
+```
+eslint foo.js
+```
+
+Will result in
+
+```
+warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override
+```
+
+It is possible to filter out ignored files using ESLint `CLIEngine`. This was the main reasoning to reject the CLI flag idea during [TSC meeting in 2018](https://gitter.im/eslint/tsc-meetings/archives/2018/08/02). The commitee was only considering the use case of integrations. However, this problem affects end users as well. Lint-staged FAQ includes a section [how can I ignore files from .eslintignore](https://github.com/okonet/lint-staged#how-can-i-ignore-files-from-eslintignore), which suggests adding additional config file and using `CLIEngine.isPathIgnored` to filter out ignored files. Having a CLI flag for this would improve the end-user experience.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+ESLint CLI should have an optional flag to disable the ignored file warning. The flag should be boolean. In order to stay consistent with current CLI flags, it should be called `--no-warning-on-ignored-files`. If it is present, ESLint should supress the `File ignored because of a matching ignore pattern.` warning. The flag should replicate `eslint.lintText`[warnIgnored option](https://eslint.org/docs/developer-guide/nodejs-api#-eslintlinttextcode-options) when it is set to `false`.
+
+The current warning should be updated to `File ignored because of a matching ignore pattern. Use "--no-ignore" to override. Use "--no-warning-on-ignored-files" to suppress this warning.`
+
+Add the flag to `optionator()` in `options.js`:
+
+```js
+{
+    option: "warning-on-ignored-files",
+    type: "Boolean",
+    default: "true",
+    description: "Prevent warnings when file list includes ignored files"
+},
+```
+
+It should be placed either unde `heading: "Miscellaneous"`, or `heading: "Handling warnings"`, or `heading: "Ignoring files"`.
+
+Add `warningOnIgnoredFiles` to `translateOptions()` in `cli.js`. It should be returned as is.
+
+Destructure `warningOnIgnoredFiles` from `options` from `internalSlotsMap.get(this)` in `executeOnFiles()` function in `cli-engine.js`.
```

Comment:
Can you also add details about changes in the [ESLint class](https://github.com/eslint/eslint/blob/main/lib/eslint/eslint.js)?

Reply:
Done. In my opinion, such implementation details should be discussed in the PR stage though.